datamodel/low/extraction_functions.go

```diff
@@ -287,11 +293,17 @@ func ExtractArray[T Buildable[N], N any](label string, root *yaml.Node, idx *ind
 			if berr != nil {
 				return nil, ln, vn, berr
 			}
+
+			if localReferenceValue == "" {
+				localReferenceValue = referenceValue
+				localIsReference = isReference
+			}
+
 			items = append(items, ValueReference[T]{
 				Value:       n,
 				ValueNode:   node,
-				IsReference: isReference,
-				Reference:   referenceValue,
+				IsReference: localIsReference,
```

Comment:
this is the fix, apologies for gofmt changing so much, I can revert the other changes if you want